### PR TITLE
[system-probe] fix compilation on clang 9

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

With Clang 9.0.1, compilation of tcp-tracer fails with error:

LLVM ERROR: Cannot select: 0x5636e19b6bf0: i64,ch = AtomicLoadOr<(load store seq_cst 8 on %ir.230)> 0x5636e1997918:1, 0x5636e19bb430, Constant:i64<0>
  0x5636e19bb430: i64 = add nuw 0x5636e1999a00, Constant:i64<8>
    0x5636e1999a00: i64,ch = CopyFromReg 0x5636e186f1b8, Register:i64 %19
      0x5636e19974a0: i64 = Register %19
    0x5636e199a698: i64 = Constant<8>
  0x5636e1999a68: i64 = Constant<0>
In function: kprobe__tcp_sendmsg

When a zero constant is passed to update_conn_stats, the compiler seems to store the 0 value in a register, that makes the code generation of __sync_fetch_and_add to fail. A `volatile` variable could be used to avoid the problem but it makes more sense to avoid the call to `__sync_fetch_and_add` with a value of zero.

### Motivation

Being able to compile the agent with Clang 9

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/ebpf/c/tracer-ebpf.c [hunk 1]
```diff
@@ -424,8 +424,10 @@ static void update_conn_stats(conn_tuple_t* t, size_t sent_bytes, size_t recv_by
 
     // If already in our map, increment size in-place
     if (val != NULL) {
-        __sync_fetch_and_add(&val->sent_bytes, sent_bytes);
-        __sync_fetch_and_add(&val->recv_bytes, recv_bytes);
+        if (sent_bytes)
```
Comment (sunhay) [inline pkg/ebpf/c/tracer-ebpf.c:427]: nit: can we make sure to add braces for these if-blocks here, even though it's only a single expression.

Why/how is LLVM registering a constant of zero for this?

Comment (lebauce) [reply pkg/ebpf/c/tracer-ebpf.c:427]: @sunhay Braces added.

The call path looks like this : kprobe__tcp_sendmsg -> handle_message(&t, size, 0) -> update_conn_stats(t, sent_bytes, recv_bytes)
In this example, the `recv_bytes` in `update_conn_stats` is the 0 constant passed from `kprobe__tcp_sendmsg`

**Review by sunhay — DISMISSED** [review]
LGTM - thanks for catching this. You'll want to get the build green 🍏 before merging!

**Comment by sunhay** [comment]
cc: @leeavital / @p-lambert `dd-gitlab/run_tests_ebpf` is failing - is it because we need to commit the new generated byte code?

**Comment by leeavital** [comment]
@sunhay  @lebauce yeah, I think so.   We need to  upgrade both the dev VM and the docker image to use clang 9.

**Comment by lebauce** [comment]
@sunhay PTAL

**Review by L3n41c — APPROVED** [review]
I confirm that this PR fixes the compilation error I have in my `agent-dev` vagrant box.

But it generates a diff in the binary blob embedded in `pkg/ebpf/bytecode/tracer-ebpf.go`.

**Review by leeavital — APPROVED** [review]

**Pull request merged**